### PR TITLE
ensure dimensionlessness of e and m0 in mono_approx

### LINKED ISSUE #1049: Units error in substitutions
This has been a recurrent problem, and I have been putting off an issue on it for a while. Now it is time. 

Look at CFP_Fuselage_Performance_int_HT.py in hoburg/d8, commit 8cac3d6. 
If you run the code, you will get the following error:
``` python
  File "c:\users\berk\dropbox (mit)\mit graduate school\code\gpkit\gpkit\nomials\nomial_math.py", line 358, in mono_lower_bound
    return self.mono_approximation(x0)
  File "c:\users\berk\dropbox (mit)\mit graduate school\code\gpkit\gpkit\nomials\nomial_math.py", line 194, in mono_approximation
    m0 *= (x0[vk])**e
  File "C:\Users\Berk\Anaconda2\lib\site-packages\pint\quantity.py", line 851, in __pow__
    raise DimensionalityError(self._units, 'dimensionless')
DimensionalityError: Cannot convert from 'force_pound' to 'dimensionless'
```
This is because of the following substitution on line 984:
```python
'W_{engine}': 11000*units('lbf'), # Engine weight substitution
```
Extremely puzzling why this doesn't work, since I have put units in other substitutions and it has worked fine. Let me know if you have any ideas @bqpd.

### PR DESCRIPTION
**Pull request by bqpd**
Closes #1049 

@whoburg, ready for review.

### CODE REVIEW

**Comment by bqpd** [comment]
test models plesae

**Comment by bqpd** [comment]
test models please

**Comment by whoburg** [comment]
sorry for the delay. planning to review this tomorrow 3/10.

**Comment by whoburg** [comment]
actually this was an easy review. looks great.

**Comment by whoburg** [comment]
add a unit test that would have been broken without this?

File: gpkit/nomials/nomial_math.py [hunk 1]
```diff
@@ -189,10 +189,13 @@ def mono_approximation(self, x0):
         m0 = 1
         for vk in self.varlocs:
             diff = self.diff(vk)
-            e = x0[vk]*mag(diff.sub(x0, require_positive=False).c)/mag(p0)
+            # we convert the x0 value to the proper units
+            x0vk = mag(x0[vk]/vk.units) if hasattr(x0[vk], "units") else x0[vk]
```
Comment (bqpd) [inline gpkit/nomials/nomial_math.py]: Use `float` instead

**Pull request merged**